hypothesis-python/src/hypothesis/internal/charmap.py

```diff
@@ -79,12 +79,20 @@ def charmap():
                 os.close(fd)
                 # Explicitly set the mtime to get reproducible output
                 with gzip.GzipFile(tmpfile, 'wb', mtime=1) as o:
-                    pickle.dump(sorted(_charmap.items()), o,
-                                pickle.HIGHEST_PROTOCOL)
+                    result = json.dumps(sorted(_charmap.items()))
+                    o.write(result.encode())
+
                 os.rename(tmpfile, f)
             except Exception:  # pragma: no cover
                 pass
     assert _charmap is not None
+
+    # each value is a tuple of 2-tuples (that is, tuples of length 2)
```

Comment:
This block of code should be indented one level further, and moved above the `assert _charmap is not None` - that way, we will only run the assertions when we first calculate the values rather than every time we use them.  I would also change the assertions to:

```python
# This line converting to tuples needs to be moved down from 
# line 73 now, to below the json.load, i.e. just before the assertions.
_charmap = {k: tuple(tuple(pair) for pair in pairs)
            for k, pairs in tmp_charmap.items()}
# Your code from here- include the comment!
for vs in _charmap.values():
    ints = list(sum(vs, ()))
    assert all([isinstance(x, int) for x in ints])
    assert ints == sorted(ints)
    assert all([len(tup) == 2 for tup in vs])
```